### ISSUE TITLE
Any way to keybind a key with multiple repeated action,such as mapping J as 5j?

### ISSUE THREAD

**Issue by amalgame21** [jarun/nnn#1445]
I used to map J and K in vim and ranger to 5j and 5k, so i can navigate up and down faster a little bit, since pageup and pagedown is jumping too fast for me and hard to control when to stop.

for vim:
`nnoremap J 5j`

for nvim (lua):
`vim.api.nvim_set_keymap('n', 'J', '5j',{ noremap = true })`

for ranger:
`map J  move down=5`

any similar thing can be done with nnn?
I tried playing around with src/nnn.h but no luck.

**Comment by jarun** [comment]
It's not available. Doesn't really work for big numbers without the numbers listed.
You can use <kbd>PgUp</kbd>, <kbd>PgDn</kbd> and <kbd>^U</kbd>, <kbd>^D</kbd>.
Also check out key <kbd>'</kbd> or set the mouse scroll lines.

**Comment by 0xACE** [comment]
Something like this

```
diff --git a/src/nnn.c b/src/nnn.c
--- src/nnn.c
+++ src/nnn.c
@@ -5928,17 +5928,20 @@
 
 static void handle_screen_move(enum action sel)
 {
 	int onscreen;
+	int offset = 1;
 
 	switch (sel) {
+	case SEL_5DOWN: offset = 5; //fallthrough
 	case SEL_NEXT:
 		if (ndents && (cfg.rollover || (cur != ndents - 1)))
-			move_cursor((cur + 1) % ndents, 0);
+			move_cursor((cur + offset) % ndents, 0);
 		break;
+	case SEL_5UP: offset = 5; //fallthrough
 	case SEL_PREV:
 		if (ndents && (cfg.rollover || cur))
-			move_cursor((cur + ndents - 1) % ndents, 0);
+			move_cursor((cur + ndents - offset) % ndents, 0);
 		break;
 	case SEL_PGDN:
 		onscreen = xlines - 4;
 		move_cursor(curscroll + (onscreen - 1), 1);
@@ -7020,8 +7023,10 @@
 				presel = FILTER;
 				clearfilter();
 			}
 			continue;
+		case SEL_5DOWN: // fallthrough
+		case SEL_5UP: // fallthrough
 		case SEL_NEXT: // fallthrough
 		case SEL_PREV: // fallthrough
 		case SEL_PGDN: // fallthrough
 		case SEL_CTRL_D: // fallthrough
diff --git a/src/nnn.h b/src/nnn.h
--- src/nnn.h
+++ src/nnn.h
@@ -49,8 +49,10 @@
 	SEL_OPEN,
 	SEL_NAV_IN,
 	SEL_NEXT,
 	SEL_PREV,
+	SEL_5DOWN,
+	SEL_5UP,
 	SEL_PGDN,
 	SEL_PGUP,
 	SEL_CTRL_D,
 	SEL_CTRL_U,
```

Wrote it on my phone via the github editor, so i have no idea if it even runs, but the gist of the changes required are there... good luck

**Comment by jarun** [comment]
@amalgame21 at commit a7ebc7858a4fb1ad4f75676562dacdbd4d3b7e6b I have added a key `J` to jump to a specific entry or relative offset from the current entry.

**Comment by amalgame21** [comment]
> Something like this
> 
> ```
> diff --git a/src/nnn.c b/src/nnn.c
> --- src/nnn.c
> +++ src/nnn.c
> @@ -5928,17 +5928,20 @@
>  
>  static void handle_screen_move(enum action sel)
>  {
>  	int onscreen;
> +	int offset = 1;
>  
>  	switch (sel) {
> +	case SEL_5DOWN: offset = 5; //fallthrough
>  	case SEL_NEXT:
>  		if (ndents && (cfg.rollover || (cur != ndents - 1)))
> -			move_cursor((cur + 1) % ndents, 0);
> +			move_cursor((cur + offset) % ndents, 0);
>  		break;
> +	case SEL_5UP: offset = 5; //fallthrough
>  	case SEL_PREV:
>  		if (ndents && (cfg.rollover || cur))
> -			move_cursor((cur + ndents - 1) % ndents, 0);
> +			move_cursor((cur + ndents - offset) % ndents, 0);
>  		break;
>  	case SEL_PGDN:
>  		onscreen = xlines - 4;
>  		move_cursor(curscroll + (onscreen - 1), 1);
> @@ -7020,8 +7023,10 @@
>  				presel = FILTER;
>  				clearfilter();
>  			}
>  			continue;
> +		case SEL_5DOWN: // fallthrough
> +		case SEL_5UP: // fallthrough
>  		case SEL_NEXT: // fallthrough
>  		case SEL_PREV: // fallthrough
>  		case SEL_PGDN: // fallthrough
>  		case SEL_CTRL_D: // fallthrough
> diff --git a/src/nnn.h b/src/nnn.h
> --- src/nnn.h
> +++ src/nnn.h
> @@ -49,8 +49,10 @@
>  	SEL_OPEN,
>  	SEL_NAV_IN,
>  	SEL_NEXT,
>  	SEL_PREV,
> +	SEL_5DOWN,
> +	SEL_5UP,
>  	SEL_PGDN,
>  	SEL_PGUP,
>  	SEL_CTRL_D,
>  	SEL_CTRL_U,
> ```
> 
> Wrote it on my phone via the github editor, so i have no idea if it even runs, but the gist of the changes required are there... good luck

I tried this code, it work great at the beginning.
However, when the remaining entry is lower than 5, it will rollover at edge an go back to the first entry even nnn was launched with `-R` flag.

**Comment by amalgame21** [comment]
> @amalgame21 at commit [a7ebc78](https://github.com/jarun/nnn/commit/a7ebc7858a4fb1ad4f75676562dacdbd4d3b7e6b) I have added a key `J` to jump to a specific entry or relative offset from the current entry.

Thats great! I tested it, it works great with the -R flag to prevent rollover at edge.
But it require input multiple keystroke  `J` `-` `5` or `J` `+` `5` everytime to move up or down once.
Any method to bind this in a key to call this function with a specific relative offset ? like press J with +5 and press K with -5?

I tried replacing the source code `char *input = xreadline(NULL, "jump (+n/-n/n): ");` with `char *input = '+5'` but seems not working. not a programmer :(

**Comment by 0xACE** [comment]
add a `min(cur+offset, ndents)` to the relevant code if rollover is a problem.

**Comment by amalgame21** [comment]
> @amalgame21 at commit [a7ebc78](https://github.com/jarun/nnn/commit/a7ebc7858a4fb1ad4f75676562dacdbd4d3b7e6b) I have added a key `J` to jump to a specific entry or relative offset from the current entry.

I replaced `char *input = xreadline(NULL, "jump (+n/-n/n): ");` with `char *input = "+5"`  "+5" double quote, and duplicate the whole `SEL_JUMP` thing to `SEL_5NEXT` and `SEL_5PREV` seems meet my need.



However there is some strange behaviour when using `-n` in this code.
When the cursor is hover on the last entry at the bottom of the page, when I press `J` `-` `1` `Enter`, the whole page scroll up, instead of only the cursor move up 1 entry.
I think the normal behaviour should be like pressing `k` once, which is the cursor moving up 1 entry, without any scrolling effect triggered.


By the way, thank you very much again, I did not expect you really actually write a new function for this minor thing for me🙏
This is much better than long pressing j and k, which make my eye very tired.
